src/js/stream_readable.js

```diff
@@ -138,6 +139,76 @@ Readable.prototype.push = function(chunk, encoding) {
 };
 
 
+Readable.prototype.pipe = function(destination, options) {
+  if (!(destination instanceof Writable || isDuplex(destination))) {
+    throw new TypeError('pipe excepts stream.Writable or' +
+                        ' stream.Duplex as argument');
+  }
+
+  options = options || {'end': true};
+
+  var listeners = {
+    'readableListener': readableListener.bind(this),
+    'dataListener': dataListener.bind(destination),
+    'endListener': endListener.bind(destination),
+  };
+
+  this.on('readable', listeners.readableListener);
+  this.on('data', listeners.dataListener);
```

Comment:
What will happen if there is a stream error? (source/destination)

Reply:
If the node.js does not state anything else, then it's ok. I'm a bit against adding extra options.